src/webview/debuggerPanel.ts

```diff
@@ -85,7 +88,16 @@ export class DebuggerPanel {
         command: 'updateStackFrames',
         stackFrames: panelViewInput.callstack.map((frame) => frame.name),
       }));
-      this.postCommandToWebViewIfViewPanelIsDefined(() => this.panelViewProxy.updatePanel(panelViewInput.callstack[0].variables));
+
+      const topFrameVariables = panelViewInput.callstack[0].variables;
+      const hasChanges = this.addToHistory(topFrameVariables);
+      if (hasChanges && (this.historyIndex === -1 || this.historyIndex >= this.inputHistory.length - 2)) {
```

Comment:
Is the index comparison with the penultimate position of the input history intended?

Reply:
Yes, because the new history entry is already added at this point (on line 93).